consensus/istanbul/validator/validator.go

```diff
@@ -63,3 +67,75 @@ func ExtractValidators(extraData []byte) []common.Address {
 
 	return addrs
 }
+
+// ConvertHashToSeed returns a random seed used to calculate proposer.
+// It converts first 7.5 bytes of the given hash to int64.
+func ConvertHashToSeed(hash common.Hash) (int64, error) {
+	// TODO-Klaytn-Istanbul: convert hash.Hex() to int64 directly without string conversion
+	hashstring := strings.TrimPrefix(hash.Hex(), "0x")
+	if len(hashstring) > 15 {
+		hashstring = hashstring[:15]
+	}
+
+	seed, err := strconv.ParseInt(hashstring, 16, 64)
+	if err != nil {
+		logger.Error("fail to make sub-list of validators", "hash", hash.Hex(), "seed", seed, "err", err)
+		return 0, err
+	}
+	return seed, nil
+}
+
+// SelectRandomCommittee composes a committee selecting validators randomly based on the seed value.
+// It returns nil if the given committeeSize is bigger than validatorSize or proposer indexes are invalid.
+func SelectRandomCommittee(validators []istanbul.Validator, committeeSize uint64, seed int64, proposerIdx int, nextProposerIdx int) []istanbul.Validator {
+	// ensure validator indexes are valid
```

Comment:
```suggestion
	// ensure proposers indexes are valid
```

Reply:
I think "validator" is good to go because the index is indexes of `validators` variable